juju/juju
# Remove fan config from model config

This removes the fanconfigurer agent facade. We no longer support
fans in 4.0 (noble/24.04 has dropped support), so we've
removed the worker. This goes one step further and this removes
the facade and the model config attribute.

We'll need to fix 3.6 to ignore the "not implemented" error when the
agent is migrated. This will be done in a follow-up PR once this
has landed.

## Checklist




## QA steps

```sh
$ juju bootstrap aws test
$ juju add-model default
$ juju deploy juju-qa-test
$ juju model-config
```

Ensure that there are no errors in the logs concerning fan/modelconfig

```sh
$ juju debug-log -m controller
$ juju debug-log
```

## Links


**Jira card:** [JUJU-6472](https://warthogs.atlassian.net/browse/JUJU-6472)



[JUJU-6472]: https://warthogs.atlassian.net/browse/JUJU-6472?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
